rust/src/cli/service.rs

```diff
@@ -74,26 +75,32 @@ pub(crate) fn ncl_service(
     Ok("".to_string())
 }
 
-// All file ending with `.yml` will be included.
+// All file ending with `.yml` that do not have a symlink ending with `.applied`
+// will be included.
 fn get_config_files(folder: &str) -> Result<Vec<PathBuf>, CliError> {
     let folder = Path::new(folder);
-    let mut ret = Vec::new();
+    let mut yml_files = HashSet::<PathBuf>::new();
+    let mut applied_files = HashSet::<PathBuf>::new();
     for entry in folder.read_dir()? {
         let file = entry?.path();
         if file.extension() == Some(OsStr::new(CONFIG_FILE_EXTENTION)) {
-            ret.push(folder.join(file));
+            yml_files.insert(folder.join(file));
+        } else if file.extension() == Some(OsStr::new(SYMLINK_FILE_EXTENTION)) {
+            applied_files.insert(folder.join(file));
```

Comment:
You need to strip the extention before insert. You current got generate `yml_files` and `applied_files` as:

```
HAHA yml_files {"/etc/nmstate/eth1_up.applied"}
HAHA applied_files {"/etc/nmstate/eth1_up.yml"}
HAHA ret ["/etc/nmstate/eth1_up.yml"]
```